### PR TITLE
refactor(client/java): remove shared tenant resources from javadocs

### LINKED ISSUE #13726: Remove shared tenant identifier again
As resource sharing is scoped out of the first iteration, we have no need for it in the Java client yet. Let's remove it from the `CommandWithTenantStep` and the `DeployResourceCommandStep2` interfaces.

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

As shared resources and a shared tenant won't be supported with multi-tenancy in Zeebe, we shouldn't document them in the client API.

:warning: This PR is based off of #13764, it should be merged after it.

## Related issues

closes #13726 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: clients/java/src/main/java/io/camunda/zeebe/client/api/command/DeployResourceCommandStep1.java [hunk 1]
```diff
@@ -107,9 +107,8 @@ interface DeployResourceCommandStep2
      *
      * <h1>Deploy resource command specifics</h1>
      *
-     * <p>When no {@code tenantId} is specified explicitly, the deployed resources are owned by all
-     * tenants using the {@link #SHARED_TENANT_IDENTIFIER} as long as multi-tenancy is enabled in
-     * the Zeebe cluster.
+     * <p>When no {@code tenantId} is specified explicitly, the deployed resources are rejected as
+     * long as multi-tenancy is enabled in the Zeebe cluster.
```
Comment (korthout) [inline clients/java/src/main/java/io/camunda/zeebe/client/api/command/DeployResourceCommandStep1.java:111]: 🔧 This is not specific to the deploy resource command. We can remove the entire override IMO.

**Review by korthout — APPROVED** [review]
Thanks @koevskinikola ❤️ 

🧹 I think we can clean this up one step further 

LGTM 👍

File: clients/java/src/main/java/io/camunda/zeebe/client/api/command/DeployResourceCommandStep1.java (hunk 1 above)
Comment (koevskinikola) [reply clients/java/src/main/java/io/camunda/zeebe/client/api/command/DeployResourceCommandStep1.java:111]: Good point, I'll remove it. If anything changes during implementation, we can adjust it later.

**Review by korthout — APPROVED** [review]
Thanks @koevskinikola 

❓ Any reason why you didn't remove the entire method overriding? 

LGTM 👍

**Comment by koevskinikola** [comment]
bors merge

**Comment by korthout** [comment]
bors retry

**Pull request merged**